packages/zoe/src/zoeService/internal-types.js

```diff
@@ -56,12 +56,14 @@
 /**
  * @callback UnwrapInstallation
  *
- * Assert the installation is known, and return the bundle and
+ * Assert the installation is known, and return the bundle/bundlecap and
  * installation
  *
  * @param {ERef<Installation>} installationP
  * @returns {Promise<{
- *   bundle: SourceBundle,
+ *   bundle?: SourceBundle,
+ *   bundleCap?: BundleCap,
+ *   bundleID?: BundleID,
```

Comment:
non-blocking but FYI, we can use unions to convey mutual exclusion. It has the advantage over optionals of guaranteeing to consumer that one of them is defined. E.g.,
```ts
type BundleOrBundleCap = {bundle: SourceBundle} | {bundleCap: BundleCap, bundleId: BundleID};
```
or in jsdoc,
```js
/** @typedef  { {bundle: SourceBundle} | {bundleCap: BundleCap: bundleId} } BundleOrBundleCap */
```

then intersection for the return type
```ts
Promise<BundleOrBundleCap & {installation: Installation}>
```

Reply:
aye, thanks, will look into that later, there's another `BundleOrBundleCap` floating around which doesn't include the `bundleID` that I don't want to collide with

Reply:
> collide with 

Want to point out that the possibility of collisions is due to ambient types. Because TypeScript is structurally typed, there's usually little problem with redundant definitions. (Besides code maintenance/documentation.)